pkg/azurefile/nodeserver.go

```diff
@@ -133,8 +172,11 @@ func (d *Driver) NodeUnpublishVolume(_ context.Context, req *csi.NodeUnpublishVo
 	volumeID := req.GetVolumeId()
 
 	klog.V(2).Infof("NodeUnpublishVolume: unmounting volume %s on %s", volumeID, targetPath)
+	if err := CleanupMountPoint(d.mounter, targetPath, true /*extensiveMountPointCheck*/); err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to unmount target %s: %v", targetPath, err)
+	}
 	if err := directvolume.Remove(targetPath); err != nil {
```

Comment:
Shouldn't we also check the runtime class here?

Reply:
no, it checks if the path is available, then clean that up. does nothing if no path is available.

Reply:
checking runtime here would not make any difference

Reply:
OK so `directvolume.Remove()` handles non-existing paths silently, sgtm.